src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreRecordReader.java

```diff
@@ -63,19 +65,43 @@ public class FirestoreRecordReader extends RecordReader<Object, QueryDocumentSna
   public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptContext)
     throws IOException, InterruptedException {
 
-    conf = taskAttemptContext.getConfiguration();
-    String projectId = conf.get(FirestoreConfig.NAME_PROJECT);
-    String serviceAccountFilePath = conf.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
-    String collection = conf.get(FirestoreConstants.PROPERTY_COLLECTION);
+    config = taskAttemptContext.getConfiguration();
+
+    String projectId = config.get(FirestoreConfig.NAME_PROJECT);
+    String databaseId = config.get(FirestoreConfig.NAME_DATABASE);
+    String collection = Strings.nullToEmpty(config.get(FirestoreConstants.PROPERTY_COLLECTION)).trim();
+
+    // Get Service Account
+    Boolean isServiceAccountFilePath = false;
+    String serviceAccountFilePath = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
+
+    // Get Service Account whether JSON or FilePath
+    String serviceAccountType = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_TYPE);
+
+    String serviceAccount = "";
+    if (serviceAccountType.equalsIgnoreCase(FirestoreConfig.SERVICE_ACCOUNT_FILE_PATH)) {
+      serviceAccount = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_FILE_PATH);
+      isServiceAccountFilePath = true;
+    } else if (serviceAccountType.equalsIgnoreCase(FirestoreConfig.SERVICE_ACCOUNT_JSON)) {
+      serviceAccount = config.get(FirestoreConfig.NAME_SERVICE_ACCOUNT_JSON);
+      isServiceAccountFilePath = false;
+    } else {
+      throw new FirestoreInitializationException("Service account type can only be either a File Path or JSON.");
+    }
```

Comment:
@itsankit-google - fyi, I forgot updating here to use the FirestoreConfig constants so updated this and then rebased it into a single commit.

Reply:
Thanks.